components/topbar/user/src/index.js

```diff
@@ -52,17 +51,25 @@ class TopbarUser extends Component {
   /**
    * Lock body element scroll.
    */
-  _lockBodyScroll () {
-    window.document.documentElement.style = HTML_DISABLED_SCROLL_STYLE
-    window.document.body.style = BODY_DISABLED_SCROLL_STYLE
+  _lockBodyScroll = () => {
+    this._verticalScrollPosition = window.scrollY
+    window.document.documentElement.classList.add('html-has-scroll-disabled')
+    window.document.body.classList.add('body-has-scroll-disabled')
+    this.props.elementsToKeepScrollOnToggleMenu.forEach(selector => {
+      document.querySelector(selector).style.transform = `translate3d(0, -${this._verticalScrollPosition}px, 0)`
```

Comment:
Don't know if it's better to save document elements in memory before.

Reply:
AFAIK it would not improve performance.
But, caching translate3d string may do.